ci/jenkins/utils.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    export GOROOT=/usr/local/go
+    export PATH=${GOROOT}/bin:$PATH
+    echo "====== Checking installed Golang version ======"
+    golang_minor_version=$(go mod edit -json | jq -r .Go)
+    antrea_golang_version="go$(go mod edit -json | jq -r .Go)"
+    installed_version=$(/usr/local/go-${golang_minor_version}/bin/go version | grep -oP "go\d+\.\d+(\.\d+)?") || true
+    # Since the antrea go.mod file does not contain the golang patch number, we will query go.dev
+    # and use the latest patch number.
```

Comment:
I think that theoretically the go.mod file can include a Go version with a patch number, but maybe that's not important in our case?

Reply:
The `go.mod` file will not include a patch number, I tried but got ` invalid go version '1.21.1': must match format 1.23`.

Reply:
Any comment on this?

Reply:
Fixed this part. Now we support both formats `go 1.20` or `go 1.21.1` in `go.mod`. If the golang version is higher than 1.21 but does not contain a patch number, this script will use `go1.21.0` as default. Otherwise, this script will just use the golang version indicated in `go.mod`(e.g. `go 1.20`, `go 1.21.2`)